magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,262 @@
+"""
+This scripts facilitates the usage of the script
+"lst1_magic_event_coincidence.py". This script is
+more like a "manager" that organizes the analysis
+process by:
+1) Creating the bash scripts for looking for
+coincidence events between MAGIC and LST in each
+night.
+2) Creating the subdirectories for the coincident
+event files.
+
+Usage:
+$ python coincident_events.py (-c config_file.yaml)
+"""
+import argparse
+import glob
+import logging
+import os
+from datetime import date as dtdt
+from datetime import timedelta
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import slurm_lines
+
+__all__ = ["configfile_coincidence", "linking_bash_lst"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configfile_coincidence(ids, target_dir, source_name, NSB_match):
+
+    """
+    This function creates the configuration file needed for the event coincidence step
+
+    Parameters
+    ----------
+    ids : list
+        List of telescope IDs
+    target_dir : str
+        Path to the working directory
+    source_name : str
+        Name of the target source
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    config_file = resource_file("config.yaml")
+    with open(
+        config_file, "rb"
+    ) as fc:  # "rb" mode opens the file in binary format for reading
+        config_dict = yaml.safe_load(fc)
+    coincidence = config_dict["event_coincidence"]
+
+    conf = {}
+    conf["event_coincidence"] = coincidence
+
+    if not NSB_match:
+        file_name = f"{target_dir}/{source_name}/config_coincidence.yaml"
+    else:
+        file_name = f"{target_dir}/v{__version__}/{source_name}/config_coincidence.yaml"
+    with open(file_name, "w") as f:
```

Comment:
as in the other file: this creating of config files can be improved: we can use directly the config file of the repository, or pass a path to the processing config file in the config_general.yaml file of the semi automatic scripts and use this instead

Reply:
We need the coincidence 'parameters' from the config.yaml in the repo and the tel. IDs from the general one. And both infos must be passed to the coincidence script that we call in conda

Reply:
yes, but why not include the path to config file (the normal one of MCP) as a parameter in config_general (the specific config file for automatic processing)?

Reply:
The one we have in the 'resource' folder? It is in a fixed (and a bit 'hidden') path, so I don't know if every user could be able to provide its path. We could allow for a default path (the one in conda/envs/....) to be used if no path is provided in the config_general

Reply:
yes, I am not sure if there is a simpler method, but I think you can do it e.g.:
```
import magicctapipe
default_path = magicctapipe.__file__.replace('/__init__.py', '/resources/config.yaml')
```